File: Python/Easy/2.py
# Challenge: Check if a Number is Prime
# Level: Beginner
# Description: This challenge involves creating a program that checks if a number entered by the user is prime or not.

# Instructions:
# 1. Ask the user to enter a number.
# 2. Create a function that takes the number as an argument.
# 3. Inside the function, check if the number is greater than 1.
# 4. If the number is less than or equal to 1, return False indicating that it is not prime.
# 5. Use a loop to iterate over all numbers from 2 to half of the entered number.
# 6. Check if the number is divisible by any of the numbers in the mentioned range.
# 7. If the number is divisible by any number in the range, return False indicating that it is not prime.
# 8. If the number is not divisible by any number in the range, return True indicating that it is prime.

# Example:
# Number: 7
# Is prime: True

# Function to check if a number is prime
def is_prime(number):
    if number <= 1:
        return False
    for i in range(2, int(number / 2) + 1):
        if number % i == 0:
            return False
    return True

# Asking user for input
user_number = int(input("Enter a number: "))

# Checking if the number is prime
result = is_prime(user_number)

# Displaying the result
print(f"Is prime: {result}")

File: PHP/Easy/2.php
<?php
// Challenge: Calculate the Sum of Digits of an Integer
// Level: Beginner
// Description: This challenge involves creating a PHP function that calculates the sum of the digits of an integer.

// Instructions:
// 1. Create a function called suma_digitos that takes an integer as a parameter.
// 2. Inside the function, initialize a variable suma to 0. This variable will store the sum of the digits.
// 3. Convert the number to a string using the strval() function.
// 4. Iterate through each digit of the string using a for or foreach loop.
// 5. In each iteration of the loop, convert the digit back to an integer using the intval() function and add its value to the suma variable.
// 6. At the end of the loop, the suma variable will contain the sum of the digits of the number.
// 7. Return the value of suma.

// Example:
// Number: 12345
// Sum of digits: 15

// Function to calculate the sum of digits of an integer
function suma_digitos($number) {
    // Initialize a variable to store the sum of digits
    $sum = 0;

    // Convert the number to a string
    $numberStr = strval($number);

    // Iterate through each digit
    for ($i = 0; $i < strlen($numberStr); $i++) {
        // Convert each digit to an integer and add it to the sum
        $sum += intval($numberStr[$i]);
    }

    // Return the sum of digits
    return $sum;
}

// Example usage:
$userNumber = readline("Enter an integer: ");
$digitSum = suma_digitos($userNumber);

echo "Sum of digits: $digitSum\n";
?>
